docs(website): improve documentation around data fetching (#1302)
commercetools/merchant-center-application-kit: website/src/content/main-concepts/data-fetching.mdx

In client-side applications, data fetching is not always easy and it usually involves a lot of boilerplate code around implementation, state management, data normalization, etc.

Since the commercetools platform has first-class support for GraphQL, we recommend to build Custom Applications using GraphQL as the main data fetching choice.

# Requests to GraphQL APIs

To handle requests to the [GraphQL APIs](/main-concepts/graphql) we use the built-in [Apollo GraphQL Client](https://www.apollographql.com/docs/react/).

<Info>

> The Apollo Client is already pre-configured within the [`<ApplicationShell>`](/main-concepts/application-shell) to connect to the `/graphql` endpoint of the [Merchant Center API Gateway](/main-concepts/api-gatway). You don't need to configure it on your own, unless you need to connect to an **external** GraphQL API (see [#connecting-to-an-external-graphql-api]).

Reviewer: Link **#connecting-to-an-external-graphql-api** is not working.

Author: Ups, my bad. Thanks!

That's it, Apollo will take care of data normalization, caching, etc.

> ## Connecting to an external GraphQL API

Author: This section is new and is based on the discussion in #1297

> In case you have your own GraphQL API, in addition to the commercetools GraphQL APIs, that is specific to your Custom Application business requirements, you want to be able to connect to it.

Reviewer: ```suggestion
In case you have a specific GraphQL API, which meets your Custom Application business requirements, it is possible to connect it in addition to the commercetools GraphQL APIs.
```

```jsx
// custom-apollo-client.js
import { createHttpLink } from 'apollo-link-http';
import { ApolloLink } from 'apollo-link';
import ApolloClient from 'apollo-client';

const httpLink = createHttpLink({
  uri: 'https://my-custom-app.com/graphql',
  headers: { accept: 'application/json' },
  fetch,
});
const link = ApolloLink.from([
  httpLink,
]);
const client = new ApolloClient({ link });

export default client;

// hello-world.js
import React from 'react';
import { useQuery } from 'react-apollo';
import Text from '@commercetools-uikit/text';
import customClient from './custom-apollo-client';
import HelloWorldQuery from './hello-world.graphql';

const HelloWorld = () => {
  const { loading, data, error } = useQuery(HelloWorldQuery, {
    client: customClient,
  });

  if (loading) return 'Loading...';
  if (error) return `Error! ${error.message}`;
  return <Text.Headline as="h1">{data.title}</Text.Headline>;
}
```

# Requests to REST APIs

Some endpoints or APIs might not be available as GraphQL but as a standard HTTP REST endpoint instead.

> To do so, you can use any HTTP client of your choice, for example the [`fetch`](https://github.github.io/fetch/) library. See [authenticating requests](/main-concepts/authentication).

Reviewer: ```suggestion
To fetch the data, you can use any HTTP client of your choice, for example the [`fetch`](https://github.github.io/fetch/) library. See [authenticating requests](/main-concepts/authentication).
```

  });
```

> Then we can use the `<Sdk.Get>` component to send the request and render the result:

Reviewer: ```suggestion
To send the request and render the result, one option is using the `<Sdk.Get>` component:
```

    render={({ isLoading, error, result }) => {
      if (isLoading) return 'Loading...';
      if (error) return `Error! ${error.message}`;
      return <Text.Headline as="h1">{`Channel: ${result.name}`}</Text.Headline>;
    }}
  />
);
export default ChannelDetails;
```

> Alternatively, you can dispatch and manage the action creator on your own, if you don't want to use the `<Sdk.Get>` component:

Reviewer: ```suggestion
Alternatively, you can dispatch and manage the action creator on your own if you don't want to use the `<Sdk.Get>` component:
```
